undertow/src/main/java/org/wildfly/extension/undertow/filters/FilterAdd.java

```diff
@@ -54,12 +52,11 @@ class FilterAdd extends AbstractAddStepHandler {
     @Override
     protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
         final String name = context.getCurrentAddressValue();
-        final ServiceTarget target = context.getServiceTarget();
-        final ServiceBuilder<?> sb = target.addService(UndertowService.FILTER.append(name));
-        final Consumer<HandlerWrapper> serviceConsumer = sb.provides(UndertowService.FILTER.append(name));
+        CapabilityServiceBuilder<?> csb = context.getCapabilityServiceTarget().addCapability(FilterCapabilities.FILTER_CAPABILITY.getDefinition());
+        Consumer<HandlerWrapper> serviceConsumer =  csb.provides(FilterCapabilities.FILTER_CAPABILITY.getDefinition(), UndertowService.FILTER.append(name));
         HandlerWrapper wrapper = this.factory.createHandlerWrapper(context, model);
-        sb.setInstance(Service.newInstance(serviceConsumer, wrapper));
-        sb.setInitialMode(ServiceController.Mode.ON_DEMAND);
-        sb.install();
+        csb.setInstance(new FilterProviderService(serviceConsumer, wrapper));
+        csb.setInitialMode(ServiceController.Mode.ACTIVE);
```

Comment:
I don't think changing the mode should be necessary for this JIRA. ON_DEMAND is the natural fit for the use of this service.

Reply:
Well, idea was to have it more pro-active about error reporting, but I dont have strong inclination to die on this hill

Reply:
FilterProviderService doesn't really do anything in start and has no dependencies, so it's not going to error.